plugins/rocksdb/src/test/java/org/hyperledger/besu/plugin/services/storage/rocksdb/unsegmented/RocksDBColumnarKeyValueStorageTest.java

```diff
@@ -73,30 +73,16 @@ public void canRemoveThroughSegmentIteration() throws Exception {
     tx.put(barSegment, bytesOf(6), bytesOf(6));
     tx.commit();
 
-    store
-        .streamKeys(fooSegment)
-        .forEach(
-            key -> {
-              if (!Arrays.equals(key, bytesOf(3))) store.tryDelete(fooSegment, key);
-            });
-    store
-        .streamKeys(barSegment)
-        .forEach(
-            key -> {
-              if (!Arrays.equals(key, bytesOf(4))) store.tryDelete(barSegment, key);
-            });
-
     for (final ColumnFamilyHandle segment : Set.of(fooSegment, barSegment)) {
-      assertThat(store.streamKeys(segment).count()).isEqualTo(1);
+      while (store.streamKeys(segment).findAny().isPresent()) {
+        store
+            .streamKeys(segment)
+            .forEach(
+                key ->
+                    assertThat(store.tryDelete(segment, key))
```

Comment:
How will this address the issue of `StorageExceptions` breaking out of the for loop?

Reply:
Oh, I think I assumed the issue was something else, let me take a look at the stacktrace carefully.